### ISSUE TITLE
No tags if "removeComments"

### ISSUE THREAD

**Issue by WolfgangKluge** [hexojs/hexo#1166]
If you set minifier-option `removeComments` to `true` you cannot use the [tag-plugins](http://hexo.io/docs/tag-plugins.html) anymore. It's all about the wrong order, I think
- replace tags with comments (`escapeContent` in hexo/lib/hexo/post.js)
- render with markdown or other renderer
- minify HTML :boom:
- replace (already removed) comments with HTML tags

The last two should be reveresed...

And there's even another issue related to this one (that could be fixed by the different order):
If you set `collapseWhitespace: true` and `conservativeCollapse: false`, there are some whitespace characters removed, that should not be removed. E.g.

```
See {% link http://hexo.io/ %} for more information.
```

Due to the wrong order, this will become
```
See <--[]0-> for more information.
```

And after html_minify
```
See<--[]0->for more information.
```
I think it's an error to remove both spaces, but [I blame html-minify for that](https://github.com/kangax/html-minifier/issues/346) - however - in that case none of the whitespaces should be removed.

So at the end I get:
```
See<a href="...">...</a>for more information.
```

Please change the order...

**Comment by Xuanwo** [comment]
- -, it looks like tommy is busy...

**Comment by naokiy** [comment]
I met same trouble.
And I tried to fixed it by #1182 . (I changed order.)
